### PR TITLE
CI: Enable wheel builds again

### LINKED ISSUE #1337: Wheels are missing on PyPI
I noticed this because of downloading dependencies and building a wheel for `asv` 0.6 in SciPy CI. It is good practice to release a wheel also for pure Python packages. No releases have one though. Is there any reason why they are missing?

### PR DESCRIPTION
**Pull request by HaoZeke**
Should close #1337. Well, once we have [TPM setup](https://docs.pypi.org/trusted-publishers/using-a-publisher/).

### CODE REVIEW

**Comment by mattip** [comment]
Please add artifact upload to this so we have a way to see the wheels before they are published

**Comment by HaoZeke** [comment]
> Please add artifact upload to this so we have a way to see the wheels before they are published
Yup, already part of it, can be viewed on the actions tab:
https://github.com/airspeed-velocity/asv/actions/runs/6134080274

I will try to add a comment action with the link.

**Comment by HaoZeke** [comment]
PyPy test failure is unrelated.

File: .github/workflows/build_wheels.yml [hunk 1]
```diff
@@ -28,7 +55,7 @@ jobs:
           - [macos-12, macosx_x86_64]
           - [macos-12, macosx_arm64]
           - [windows-2019, win_amd64]
-        python-version: ['3.7', '3.8', '3.9', '3.10', '3.11']
+        python-version: ['3.8', '3.9', '3.10', '3.11', 'pypy3.8', 'pypy3.9']
```
Comment (mattip) [inline .github/workflows/build_wheels.yml:58]: If PyPy3.8 causes problems, we should drop it.

**Comment by mattip** [comment]
Thanks @HaoZeke

**Pull request merged**